marlowe-playground-client/src/ConfirmUnsavedNavigation/State.purs

```diff
@@ -0,0 +1,43 @@
+module ConfirmUnsavedNavigation.State where
+
+import Halogen.HTML
+import ConfirmUnsavedNavigation.Types (Action(..), State, _wantsToSaveProject)
+import Data.Lens (assign)
+import Data.Maybe (Maybe(..))
+import Effect.Aff.Class (class MonadAff)
+import Effect.Class.Console (log)
+import Halogen (HalogenM, liftEffect)
+import Halogen.HTML.Events (onClick)
+import Halogen.HTML.Properties (classes)
+import MainFrame.Types (ChildSlots)
+import Marlowe (SPParams_)
+import Prelude (Unit, Void, const, ($), discard)
+import Servant.PureScript.Settings (SPSettings_)
+
+handleAction ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  Action -> HalogenM State Action ChildSlots Void m Unit
+handleAction settings DontSaveProject = do
+  liftEffect $ log $ "setting wants to save to false"
```

Comment:
I think you should always use `Debug.trace` since we don't actually want to use the log in "production". Debug leaves a compiler warning which is preferable.